soh/soh/Enhancements/randomizer/3drando/hints.cpp

```diff
@@ -765,6 +770,26 @@ void CreateDampesDiaryText() {
   dampesText = temp1 + area + temp2;
 }
 
+void CreateGregRupeeHint() {
+  uint32_t item = 0;
+  uint32_t location = FilterFromPool(allLocations, [item](const uint32_t loc){return Location(loc)->GetPlacedItemKey() == GREG_RUPEE;})[0];
```

Comment:
based on what we have here <https://github.com/HarbourMasters/Shipwright/pull/2530/files>

```suggestion
  uint32_t location = FilterFromPool(allLocations, [](const uint32_t loc){return Location(loc)->GetPlacedItemKey() == GREG_RUPEE;})[0];
```